### PR TITLE
feat(eda) : add a drop_null function

### PR DESCRIPTION
**Pull request by peterirani**
Add a function to remove NA/Null values from a Pandas/Dask dataframe or series.

# Description
Tried to fix issue #180 . Added a custom function to drop the NA/NULL values from a Pandas/Dask DataFrame or Series. 

# How Has This Been Tested?
This was tested on the Titanic dataset (using both .csv and xlsx file types).

# Snapshots:
Include snapshots for easier review.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have already squashed the commits and make the commit message conform to the project standard.
- [x] I have already marked the commit with "BREAKING CHANGE" or "Fixes #" if needed.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by peterirani** [comment]
Hi Brandon, I have made the necessary changes. Can you please have a look at them.

File: dataprep/eda/basic/compute.py [hunk 1]
```diff
@@ -331,12 +332,15 @@ def compute_univariate(
...
         # qq plot
-        qqdata = calc_qqnorm(df[x].dropna())
+        qqdata = calc_qqnorm(df_x)
         # kde plot
-        kdedata = calc_hist_kde(df[x].dropna().values, bins)
+        kdedata = calc_hist_kde(df_x.values, bins)
         # box plot
-        boxdata = calc_box(df[[x]].dropna(), bins, dtype=dtype)
+        boxdata = calc_box(df_x, bins, dtype=dtype)
         # histogram
```
Comment (brandonlockhart) [inline dataprep/eda/basic/compute.py:344]: `df_x` is a series but `df[[x]].dropna()` is a dataframe, and `calc_box` only takes a dataframe as input. Can you please do `df_x.to_frame()` and do some testing to make sure it works?

File: dataprep/eda/dtypes.py [hunk 2]
```diff
@@ -233,3 +255,26 @@ def is_pandas_categorical(dtype: Any) -> bool:
...
+    if isinstance(var, x.__args__):
+        return var[~var.isin(NULL_VALUES)]
+
+    elif isinstance(var, y.__args__):
+        df = var
+        for values in var.columns:
+            df = df[~df[values].isin(NULL_VALUES)]
+        return df
+
+    return None
```
Comment (brandonlockhart) [inline dataprep/eda/dtypes.py:280]: Can you please change this function to

``` python

def drop_null(
    var: Union[pd.Series, dd.Series, pd.DataFrame, dd.DataFrame]
) -> Union[pd.Series, dd.Series, pd.DataFrame, dd.DataFrame]:
    """
    Drop the null values (specified in NULL_VALUES) from a series or DataFrame
    """

    if isinstance(var, (pd.Series, dd.Series)):
        if is_datetime(var.dtype):
            return var.dropna()
        return var[~var.isin(NULL_VALUES)]

    elif isinstance(var, (pd.DataFrame, dd.DataFrame)):
        df = var
        for values in df.columns:
            if is_datetime(df[values].dtype):
                df = df.dropna(subset=[values])
            else:
                df = df[~df[values].isin(NULL_VALUES)]
        return df

    return None
```

`isin` does not work for datetime columns. However, if a column is datetime with errors, they will necessarily be marked as NaT, therefore, `dropna()` will remove all values that don't match the specified datetime format in the column.

File: dataprep/eda/basic/compute.py (hunk 1 above)
Comment (peterirani) [reply dataprep/eda/basic/compute.py]: made the changes!

File: dataprep/eda/dtypes.py (hunk 2 above)
Comment (peterirani) [reply dataprep/eda/dtypes.py]: Thanks for the pointing this out! I have updated the function.

**Comment by peterirani** [comment]
For `poetry run pytest dataprep` it fails on dataprep/tests/data_connector/test_integration.py and gives the following error **KeyError: 'DATAPREP_DATA_CONNECTOR_YOUTUBE_TOKEN'** I tried the same command by downloading the dataprep folder and it gave the same error.  

I am also attaching the screenshot below:
Screen Shot 2020-07-12 at 9 24 19 PM

**Comment by peterirani** [comment]
Hi Brandon, I have incorporated all the changes, the only test failing are the Data_connector tests. Can you please have a look at this and merge this.

**Comment by dovahcrow** [comment]
> For `poetry run pytest dataprep` it fails on dataprep/tests/data_connector/test_integration.py and gives the following error **KeyError: 'DATAPREP_DATA_CONNECTOR_YOUTUBE_TOKEN'** I tried the same command by downloading the dataprep folder and it gave the same error.
> 
> I am also attaching the screenshot below:
> Screen Shot 2020-07-12 at 9 24 19 PM

Hi Peter,  you can set `DATAPREP_SKIP_CREDENTIAL_TESTS` to 1 to skip these tests that require credentials.

File: dataprep/eda/dtypes.py [hunk 3]
```diff
@@ -233,3 +255,27 @@ def is_pandas_categorical(dtype: Any) -> bool:
...
+    elif isinstance(var, (pd.DataFrame, dd.DataFrame)):
+        df = var
+        for values in df.columns:
+            if is_datetime(df[values].dtype):
+                df = df.dropna(subset=[values])
+            else:
+                df = df[~df[values].isin(NULL_VALUES)]
+        return df
+
+    return None
```
Comment (dovahcrow) [inline dataprep/eda/dtypes.py:281]: It would be better to make the error explicit by raising an exception.

Comment (peterirani) [reply dataprep/eda/dtypes.py]: Hi Young, I added a ValueError with a custom message. For setting `DATAPREP_SKIP_CREDENTIAL_TESTS` to 1, in what file is this variable? test_integration.py?

Comment (dovahcrow) [reply dataprep/eda/dtypes.py]: You can just set it in your environment variables

**Comment by dovahcrow** [comment]
lgtm. Just let it be for codecov.

**Pull request merged**